src/components/vm/disks/diskAdd.jsx

```diff
@@ -365,511 +358,511 @@ const CustomPath = ({ idPrefix, onValueChanged, device, validationFailed, hideDe
+            {defaultBody}
+        </Modal>
+    );
+};
+
+const AddDiskModalFooter = ({
+    customDiskVerificationFailed,
+    customDiskVerificationMessage,
+    dialogErrorSet,
+    dialogLoading,
+    disk,
+    diskParams,
+    idPrefix,
+    isMediaInsertion,
+    mode,
+    setValidationFailed,
+    storagePoolName,
+    storagePools,
+    verificationInProgress,
+    vm,
+    vms,
+}) => {
+    const [addDiskInProgress, setAddDiskInProgress] = useState(false);
+    const Dialogs = useDialogs();
 
-    onInsertClicked() {
-        const Dialogs = this.context;
-        const close = Dialogs.close;
-        const { vm } = this.state;
-        const { disk } = this.props;
+    const validateParams = () => {
+        const validationFailed = {};
+        const storagePoolType = diskParams.storagePool?.type;
+
+        if (mode !== CUSTOM_PATH && !storagePoolName)
+            validationFailed.storagePool = _("Please choose a storage pool");
+
+        if (mode === CUSTOM_PATH && customDiskVerificationFailed)
+            validationFailed.customPath = customDiskVerificationMessage;
+
+        if (mode === CREATE_NEW) {
+            if (!diskParams.volumeName) {
+                validationFailed.volumeName = _("Please enter new volume name");
+            }
+            if (poolTypesNotSupportingVolumeCreation.includes(storagePoolType)) {
+                validationFailed.storagePool = cockpit.format(_("Pool type $0 does not support volume creation"), storagePoolType);
+            }
+            const poolCapacity = parseFloat(convertToUnit(storagePools.find(pool => pool.name == storagePoolName).capacity, units.B, diskParams.unit));
+            if (diskParams.size > poolCapacity) {
+                validationFailed.size = cockpit.format(_("Storage volume size must not exceed the storage pool's capacity ($0 $1)"), poolCapacity.toFixed(2), diskParams.unit);
+            }
+        } else if (mode === USE_EXISTING) {
+            if (mode !== CUSTOM_PATH && !diskParams.existingVolumeName)
+                validationFailed.existingVolumeName = _("Please choose a volume");
+        }
 
-        const validation = this.validateParams();
-        if (Object.getOwnPropertyNames(validation).length > 0)
-            return this.setState({ addDiskInProgress: false, validate: true });
+        return validationFailed;
+    };
 
-        this.setState({ addDiskInProgress: true, validate: false });
+    const onInsertClicked = () => {
         return domainInsertDisk({
             connectionName: vm.connectionName,
             vmName: vm.name,
             target: disk.target,
-            diskType: this.state.mode === CUSTOM_PATH ? "file" : "volume",
-            file: this.state.file,
-            poolName: this.state.storagePoolName,
-            volumeName: this.state.existingVolumeName,
+            diskType: mode === CUSTOM_PATH ? "file" : "volume",
```

Comment:
This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-1303-20231110-154059-d19bea72-fedora-39-devel//Coverage/lcov/github-pr.diff.gcov.html)